control/metrics.go

```diff
@@ -55,17 +55,28 @@ var (
 
 // hostnameReader, hostnamer created for mocking
 func init() {
-	hostnameReader = &hostnameReaderType{}
+	h, err := os.Hostname()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"_module": "control",
+			"_file":   "metrics.go,",
+			"_block":  "addStandardAndWorkflowTags",
+			"error":   err.Error(),
+		}).Error("Unable to determine hostname")
+	}
+	hostnameReader = &hostnameReaderType{hostname: h}
```

Comment:
I think requiring a restart is nasty and we should avoid that one.

Maybe having a TTL that is pretty long (60min)? And have it invalidate and refresh past that. You can just have the accessor function hide this behavior. This way we prevent too much churn (imagine a 10ms interval) but prevent hard locked restart.

We would just need to add comments to the README to call out the 60min refresh. We could postpone exposing that as config until someone asks for it.